fdb-relational-core/src/main/java/com/apple/foundationdb/relational/api/RowStruct.java

```diff
@@ -333,7 +336,8 @@ public String toString() {
         } catch (SQLException e) {
             throw new UncheckedRelationalException(new RelationalException(e));
         }
-
+        // restore wasNull
+        wasNull = preservedWasNullValue;
```

Comment:
Could you restore wasNull in a finally block? 
To make sure that wasNull is not wrong if there is an exception being thrown.